cpp/src/gandiva/to_date_holder.cc

```diff
@@ -51,7 +50,7 @@ Status ToDateHolder::Make(const FunctionNode& node,
   if (node.children().size() == 3) {
     auto literal_suppress_errors =
         dynamic_cast<LiteralNode*>(node.children().at(2).get());
-    if (literal_pattern == nullptr) {
+    if (literal_suppress_errors == nullptr) {
```

Comment:
This is a bug I found accidentally during the refactoring. The `literal_pattern` is always non `nullptr` since line 37 already check if it is `nullptr`. And the code here would like to check the third parameter, which should be `literal_suppress_errors` I believe